lib/mozart_fetcher/component.ex

```diff
@@ -1,42 +1,80 @@
 defmodule MozartFetcher.Component do
-  alias MozartFetcher.{Component, Config, Envelope, LocalCache}
-
-  @derive [Jason.Encoder]
-  defstruct [:index, :id, :status, :envelope]
+  alias MozartFetcher.{Config, Envelope, LocalCache}
 
   def fetch({config = %Config{}, component_index}) do
     process(component_index, config, get(config))
   end
 
+  defp process(component_index, config = %Config{format: "ares"}, {:ok, %HTTPoison.Response{status_code: 200, body: body}}) do
+    case Jason.decode(body, keys: :atoms) do
+      {:ok, data} ->
+        metric(config.id, 200)
+        %{index: component_index, id: config.id, status: 200, data: data}
+      {:error, _err} ->
+        metric(config.id, :json_decode_error)
+        failed_component(component_index, :json_decode_error, config.id, :ares)
+    end
+  end
+
   defp process(component_index, config, {:ok, %HTTPoison.Response{status_code: 200, body: body}}) do
-    ExMetrics.increment("success.component.process")
-    ExMetrics.increment("success.component.process.#{config.id}.200")
-    %Component{index: component_index, id: config.id, status: 200, envelope: Envelope.build(body)}
+    metric(config.id, 200)
+    %{index: component_index, id: config.id, status: 200, envelope: Envelope.build(body)}
   end
 
-  defp process(component_index, config, {:ok, %HTTPoison.Response{status_code: status_code, body: body}}) do
-    ExMetrics.increment("error.component.process")
-    ExMetrics.increment("error.component.process.#{config.id}.#{status_code}")
-    Stump.log(:error, %{message: "Non-200 response. Got status:#{status_code} for component #{config.id}"})
-    %Component{index: component_index, id: config.id, status: status_code, envelope: %Envelope{}}
+  defp process(component_index, config = %Config{format: "ares"}, {:ok, %HTTPoison.Response{status_code: status_code}}) do
+    metric(config.id, status_code)
+    %{index: component_index, id: config.id, status: status_code, data: %{}}
+  end
+
+  defp process(component_index, config, {:ok, %HTTPoison.Response{status_code: status_code}}) do
+    metric(config.id, status_code)
+    %{index: component_index, id: config.id, status: status_code, envelope: %Envelope{}}
+  end
+
+  defp process(component_index, config = %Config{format: "ares"}, {:error, %HTTPoison.Error{reason: reason}}) do
+    metric(config.id, reason)
```

Comment:
What can reason be? If it's a message, this could be long in the metric name.

Reply:
An atom with a (usually one word) reason why the connection failed. I couldn't find a full list but the doccumentation mentions `:closed`, `:econnrefused`, and we see `:timeout` in the tests and in prod

Reply:
I wonder if we should either hardcode "error" or use a status. We have "component-200" for successful ones, so we could have "component-404" or "component-500" when they are non-200 responses.

Otherwise to see how many errors of a particular component, we'd have a lot to add in Grafana.

Reply:
We already have status code ones for non 200 responses. These errors happen when we don't even receive a status code. I believe we will still want to see metrics for these.

I think the solution is not fewer, and more generic metrics, but rather just adding more metrics.
We should keep the `error.component.process.#{id}.#{status}"` and `error.component.process.#{id}.#{reason}"` but also include a `error.component.process.#{id}`, `error.component.process.#{reason}"`, and `error.component.process.#{status}"` metric so we can graph different situations better

Reply:
I've just pushed a commit adding these new alarms so we can create all the graphs and alarms as we need